### PR TITLE
Add systemd timer to update root.hints file

### PR DESCRIPTION
**Pull request by bastelfreak**
By default we download a root.hints file once. That's bad. it contains IP-addresses for all root DNS servers. Those addresses can change from time to time. We should update the file every now and then.

### CODE REVIEW

**Review by rwaffen — APPROVED** [review]

File: templates/roothints.service.epp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<%- | Stdlib::Absolutepath $hints_file, Stdlib::HTTPSUrl $root_hints_url| -%>
+# THIS FILE IS MANAGED BY PUPPET
+# BASED ON https://wiki.archlinux.org/title/Unbound#Roothints_systemd_timer
+[Unit]
+Description=Update root hints for unbound
+After=network.target
+
+[Service]
+ExecStart=/usr/bin/curl --output <%= $hints_file %> <%= $root_hints_url %>
```
Comment (evgeni) [inline templates/roothints.service.epp:9]: I guess this should use `unbound::fetch_client` and not `curl`?

**Review by evgeni — CHANGES_REQUESTED** [review]

File: templates/roothints.service.epp (hunk 1 above)
Comment (bastelfreak) [reply templates/roothints.service.epp:9]: I considered this, but! `unbound::fetch_client` doesn't have an absolute path and `ExecStart` requires this :( Should we use `ExecStart=/usr/bin/$unbound::fetch_client`? That's quite ugly, but it seems to be fine on CI.

Comment (evgeni) [reply templates/roothints.service.epp]: it's better than using `curl` here and `wget` there :)

also, I thought recent systemd is OK with relative paths here?

Comment (bastelfreak) [reply templates/roothints.service.epp]: okay I tested it locally and it works with a relative path. let's see what CI says.

Comment (evgeni) [reply templates/roothints.service.epp]: Seems EL7 is unhappy about that

Comment (bastelfreak) [reply templates/roothints.service.epp]: I hate old software so much

File: manifests/init.pp [hunk 2]
```diff
@@ -316,6 +319,24 @@
       mode    => '0444',
       content => $hints_file_content,
     }
+    if $update_root_hints {
```
Comment (ekohl) [inline manifests/init.pp:322]: Should it be able to ensure the timer is absent?

Comment (bastelfreak) [reply manifests/init.pp]: implemented it

File: templates/roothints.service.epp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<%- | Stdlib::Absolutepath $hints_file, Stdlib::HTTPSUrl $root_hints_url, String[1] $fetch_client | -%>
+# THIS FILE IS MANAGED BY PUPPET
+# BASED ON https://wiki.archlinux.org/title/Unbound#Roothints_systemd_timer
+[Unit]
+Description=Update root hints for unbound
+After=network.target
+
+[Service]
+ExecStart=<%= $fetch_client %> --output <%= $hints_file %> <%= $root_hints_url %>
```
Comment (evgeni) [inline templates/roothints.service.epp:9]: Fetch client already includes -O

**Comment by kenyon** [comment]
I don't run DNS anymore, but when I did, I'd just use `/usr/share/dns/root.hints` that is installed by the [dns-root-data](https://tracker.debian.org/pkg/dns-root-data) package.

**Comment by bastelfreak** [comment]
Also a nice idea, but that seems to be very Debian specific. Unbound also has hints compiled into it that can be used.

**Review by evgeni — APPROVED** [review]

**Pull request merged**